oximeter/impl/src/schema/mod.rs

```diff
@@ -153,18 +159,58 @@ fn validate_timeseries_name(s: &str) -> Result<&str, MetricsError> {
     }
 }
 
+/// Text descriptions for the target and metric of a timeseries.
+#[derive(Clone, Debug, Default, Deserialize, JsonSchema, Serialize)]
+pub struct TimeseriesDescription {
+    pub target: String,
+    pub metric: String,
+}
+
+/// Measurement units for timeseries samples.
+#[derive(Clone, Copy, Debug, Deserialize, Serialize, JsonSchema, PartialEq)]
+#[serde(rename_all = "snake_case")]
+// TODO-completeness: Include more units, such as power / temperature.
+// TODO-completeness: Decide whether and how to handle dimensional analysis
+// during queries, if needed.
+pub enum Units {
+    Count,
+    Bytes,
+}
+
+impl Units {
+    /// Construct a `Units::Count`.
+    pub const fn count() -> Self {
+        Units::Count
+    }
+}
+
 /// The schema for a timeseries.
 ///
 /// This includes the name of the timeseries, as well as the datum type of its metric and the
 /// schema for each field.
 #[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
 pub struct TimeseriesSchema {
     pub timeseries_name: TimeseriesName,
+    // TODO-cleanup: This default should be removed once schema are tracked in
+    // CockroachDB. Same for the version and scope fields.
+    #[serde(default)]
+    pub description: TimeseriesDescription,
```

Comment:
How is this related to `timeseries_name`? If it's expected to always match exactly, could we drop it entirely and just use `timeseries_name` as needed?

Reply:
Ah, I see the confusion of some earlier comments. The _description_ is a freeform textual description of what the timeseries is supposed to measure, with a component for the target and metric. That provides extra context and details, beyond the existing _name_. It's a docstring, basically.